OTOnexus/Classes/Capture/OTOCaptureView.swift

```diff
@@ -407,9 +402,48 @@ extension OTOCaptureView {
             }
         #endif
         
+        return corners
+    }
+    
+    fileprivate func barcodeOrientation() -> AVCaptureVideoOrientation {
+        guard let scannedCodeObject = scannedCodeObject else {
+            return .portrait
+        }
+        let corners = self.corners(for: scannedCodeObject)
+        
+        if corners.count == 4 {
+            let corner1 = corners[0]
+            let corner3 = corners[2]
+            if abs(corner1.x - corner3.x) > abs(corner1.y - corner3.y) {
+                if(corner1.x < corner3.x) {
+                    return .landscapeRight
+                } else {
+                    return .landscapeLeft
+                }
+            } else {
+                if(corner1.y < corner3.y) {
+                    return .portraitUpsideDown
+                } else {
+                    return.portrait
+                }
+            }
+        }
+        
+        return .portrait
+    }
+    
+    fileprivate func pathForCorners(transformedForLayer previewLayer: AVCaptureVideoPreviewLayer) -> CGPath {
+        guard let barcodeObject = scannedCodeObject else { return CGMutablePath() }
+        let transformedObject = previewLayer.transformedMetadataObject(for: barcodeObject) as? AVMetadataMachineReadableCodeObject
+        
+        // new mutable path
+        let path = CGMutablePath()
+        
+        let corners = self.corners(for: transformedObject)
+        
         for corner in corners {
-            path.move(to: corner)
             path.addLine(to: corner)
+            path.move(to: corner)
```

Comment:
I wonder if this is even necessary at all, but the way I usually build paths is to move to a point, then add lines to all the other points. So, I'd think we would need one call to `move(to:)` _before_ the for loop starts. But it appears it's working without that, so there must be something else going on.

Reply:
so I tested it and I thought it was working